indico/modules/rb/client/js/components/ClipboardButton.jsx

```diff
@@ -0,0 +1,58 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2020 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import React from 'react';
+
+import PropTypes from 'prop-types';
+import {Popup, Icon} from 'semantic-ui-react';
+import {Translate} from 'indico/react/i18n';
+
+export default class ClipboardButton extends React.Component {
+  static propTypes = {
+    text: PropTypes.string.isRequired,
+  };
+
+  state = {
+    copiedLink: false,
+  };
+
+  handleOpen = () => {
+    this.setState({copiedLink: true});
+
+    this.timeout = setTimeout(() => {
+      this.setState({copiedLink: false});
+    }, 2000);
+  };
+
+  handleClose = () => {
+    this.setState({copiedLink: false});
+    clearTimeout(this.timeout);
+  };
+
+  render() {
+    const {text} = this.props;
+    const {copiedLink} = this.state;
+
+    return (
+      <Popup
+        on="click"
+        position="bottom center"
+        trigger={
+          <Icon
+            style={{cursor: 'pointer'}}
+            name="linkify"
+            onClick={() => navigator.clipboard.writeText(text)}
```

Comment:
I think we still show the popup here even if copying was delayed (browsers might ask for consent to access the clipboard) since we don't await the promise returned by writeText. You can probably do the copying in onOpen and only set a new `open` state flag (used in the Popup's `open` prop) after you finished copying.